src/model.ts

```diff
@@ -7,67 +7,124 @@ import { IContentFieldFilter, IContentSearchRequest } from '@esri/hub-search';
 import { PassThrough } from 'stream';
 import { PagingStream } from './paging-stream';
 import { getBatchedStreams } from './helpers/get-batched-streams';
-import { fetchSite, getHubApiUrl, getPortalApiUrl, hubApiRequest, RemoteServerError } from '@esri/hub-common';
+import { fetchSite, getHubApiUrl, getPortalApiUrl, hubApiRequest, IModel, RemoteServerError } from '@esri/hub-common';
+
+const REQUIRED_FIELDS = [
+  'id', // used for the dataset landing page URL
+  'type', // used for the dataset landing page URL
+  'slug', // used for the dataset landing page URL
+  'access', // used for detecting proxied csv's
+  'size', // used for detecting proxied csv's
+  'licenseInfo', // required for license resolution
+  'structuredLicense', // required for license resolution
+];
+
+// additional fields due to dataset enrichment
+const ADDON_FIELDS = [
+  'hubLandingPage',
+  'accessUrlCSV',
+  'isLayer',
+  'accessUrlKML',
+  'accessUrlShapeFile',
+  'accessUrlWFS',
+  'accessUrlWMS',
+  'accessUrlGeoJSON',
+  'license'
+];
+
+type HubApiRequest = {
+  hostname: string,
+  res?: {
+    locals: {
+      searchRequestBody?: IContentSearchRequest,
+      siteModel?: IModel,
+      arcgisPortal?: string
+    }
+  }
+  query?: {
+    limit?: number
+  }
+}
 
 export class HubApiModel {
 
-  async getStream (request: Request) {
-    const searchRequest: IContentSearchRequest = request.res.locals.searchRequest;
-    this.preprocessSearchRequest(searchRequest);
+  async getStream(request: Request) {
+    const { hostname, res: { locals: { searchRequestBody, siteModel, arcgisPortal } }, query: { limit } }: HubApiRequest = request;
```

Comment:
`hostname` shouldn't come from `request.hostname` here.  It's only our hub-feeds koop instance will have a request.hostname that should be used as the hub-site identifier. someone else may want to use the search-provider in a koop instance, and they may just want pass in a site-identifier as a query parameter, etc.

When we get to the provider, anything we need should be in `locals` or on the request.query.  In our hub-feeds instance, we should extract the hostname and add it a property `siteIdentifier` in `locals`.

Reply:
Added `siteIdentifier` in koop instance and replaced hostname with `siteIdentifier` from `locals` in the provider.